ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Byron/Forge.hs

```diff
@@ -119,15 +145,35 @@ forgeBlock (WithEBBNodeConfig cfg) curSlot curNo prevHash txs isLeader = do
       , pbftProtocolMagic
       } = encNodeConfigExt cfg
 
+    blockPayloads :: BlockPayloads
+    blockPayloads = foldl' extendBlockPayloads initBlockPayloads txs
+
     txPayload :: CC.UTxO.TxPayload
-    txPayload = CC.UTxO.mkTxPayload (map (void . byronTx) txs)
+    txPayload = CC.UTxO.mkTxPayload (bpTxs blockPayloads)
+
+    dlgPayload :: CC.Delegation.Payload
+    dlgPayload = CC.Delegation.unsafePayload (bpDlgCerts blockPayloads)
```

Comment:
Is there safe way to construct such a payload?

Reply:
I don't think so. This seems to be the way that delegation payloads are constructed throughout `cardano-ledger`. But I don't think this is a big deal as we're just constructing the payload here and it'll actually be validated later on.

Reply:
No, not as far as I know. `unsafePayload` or the constructor `UnsafeAPayload` are what seem to be used throughout `cardano-ledger`.